### DISCUSSION TITLE
shebang sniffing in nushell

### DISCUSSION THREAD

**Discussion by shemeshg** [nushell/nushell#5922] [Q&A] [answered]
# Seems nushell assumes /bin/sh always.

## hello.py
```python
#!/usr/local/bin/python3
print("hello world")
```
`./hello.py` will not work

## hello.sh
```bash
#!/bin/sh
python3 <<END
print("hello world")
END
```

`./hello.sh` this **work ok**.

## hello.py with `#!/usr/bin/env` also **work ok**
```python
#!/usr/bin/env python3
print("hello world")
```

Is it intensionally?

**Comment by fdncred** [comment, marked as answer]
correct. there is no shebang sniffing in nushell.
